src/ImageSharp/Processing/Processors/Binarization/BinaryThresholdProcessor.cs

```diff
@@ -5,6 +5,22 @@
 
 namespace SixLabors.ImageSharp.Processing.Processors.Binarization
 {
+    /// <summary>
+    /// The color component to be compared to threshold.
+    /// </summary>
+    public enum BinaryThresholdValueType : int
```

Comment:
`BinaryThresholdComponentKind` ? Naming is hard...

Reply:
Oh yes, that's true.
My naming candidates have been
- BinaryThresholdColorComponent
- BinaryThresholdValue
- BinaryThresholdValueType
Now Jim added
- BinaryThresholdComponentKind
My intention would be to make the enumeration expandable for each color component of each color model.
I'm open regarding names. How can we find the best one?

Reply:
What about `BinaryThresholdMethod`?

Reply:
If I get it right, it's not a method, it's just just a value selector.

Reply:
`BinaryThresholdComponentKind` feels better than `BinaryThresholdValueType`, but based on the semantics you described, my favorite from the list above is `BinaryThresholdColorComponent`. I don't think we need a postfix like `*Type` or `*Kind` when the name already ends to `*Component`. @JimBobSquarePants thoughts?

Reply:
Yeah... `BinaryThresholdComponent` works well. We've managed to avoid the `**Kind` suffix elsewhere.